include/ada/helpers.h

```diff
@@ -209,23 +209,23 @@ inline int leading_zeroes(uint32_t input_num) noexcept {
  * faster than std::to_string(x).size().
  * @return digit count
  */
-inline int fast_digit_count(uint32_t x) noexcept {
-  auto int_log2 = [](uint32_t z) -> int {
-    return 31 - ada::helpers::leading_zeroes(z | 1);
+inline int fast_digit_count(uint32_t const x) noexcept {
+  auto int_log2 = [](uint32_t const z) -> int {
+    return 31 - ada::helpers::leading_zeroes(z | 1U);
   };
   // Compiles to very few instructions. Note that the
   // table is static and thus effectively a constant.
   // We leave it inside the function because it is meaningless
   // outside of it (this comes at no performance cost).
-  const static uint64_t table[] = {
+  constexpr static uint64_t table[] = {
       4294967296,  8589934582,  8589934582,  8589934582,  12884901788,
       12884901788, 12884901788, 17179868184, 17179868184, 17179868184,
       21474826480, 21474826480, 21474826480, 21474826480, 25769703776,
       25769703776, 25769703776, 30063771072, 30063771072, 30063771072,
       34349738368, 34349738368, 34349738368, 34349738368, 38554705664,
       38554705664, 38554705664, 41949672960, 41949672960, 41949672960,
       42949672960, 42949672960};
-  return int((x + table[int_log2(x)]) >> 32);
+  return static_cast<int>((x + table[int_log2(x)]) >> 32U);
```

Comment:
Here you are 'correcting' 

```C++
return int((x + table[int_log2(x)]) >> 32);
```
and replacing it with 
```
static_cast<int>((x + table[int_log2(x)]) >> 32U);.
```

When would the more complicated version you propose be better/more correct. Can you give an example where the two expressions would give different results?

Reply:
`int(...)` is a c-style cast. It is impossible to find with a grep. The "intent" is to cast to int, why not express that intent?

The `U` suffix is there because **it's not a good idea to use signed integers in bitwise operations**, even though it has no effect here.